### PR TITLE
pdfannots2json: init at 1.0.16

### PR DESCRIPTION
**Pull request by 0nyr**
## Description of changes

This package called `pdfannots2json` is used for Obsidian <-> Zotero bridging, by the popular Zotero Integration library with more that 196000 downloads. This package does't work by default on nixos as it requires Go to work.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pd/pdfannots2json/package.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
```
Comment (hughobrien) [inline pkgs/by-name/pd/pdfannots2json/package.nix:3]:
```suggestion
buildGoModule {
```

`rec` might not be necessary

File: pkgs/by-name/pd/pdfannots2json/package.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "pdfannots2json";
+  version = "1.0.16";
+
+  src = fetchFromGitHub {
+    owner = "mgmeyers";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-qk4OSws/6SevN/Q0lsyxw+fZkm2uy1WwOYYL7CB7QUk=";  # Replace with the correct sha256
```
Comment (hughobrien) [inline pkgs/by-name/pd/pdfannots2json/package.nix:11]:
```suggestion
    sha256 = "sha256-qk4OSws/6SevN/Q0lsyxw+fZkm2uy1WwOYYL7CB7QUk=";
```

File: pkgs/by-name/pd/pdfannots2json/package.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "pdfannots2json";
+  version = "1.0.16";
+
+  src = fetchFromGitHub {
+    owner = "mgmeyers";
+    repo = pname;
```
Comment (hughobrien) [inline pkgs/by-name/pd/pdfannots2json/package.nix:9]:
```suggestion
    repo = "pdfannots2json;
```

See here https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/by-name/pd/pdfannots2json/package.nix (hunk 1 above)
Comment (0nyr) [reply pkgs/by-name/pd/pdfannots2json/package.nix:3]: Ok, thanks for the review.

File: pkgs/by-name/pd/pdfannots2json/package.nix (hunk 3 above)
Comment (0nyr) [reply pkgs/by-name/pd/pdfannots2json/package.nix:9]: Ok, thanks.

**Comment by hughobrien** [comment]
Result of `nixpkgs-review pr 284847` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pdfannots2json</li>
  </ul>
</details>

**Review by hughobrien — APPROVED** [review]

**Comment by hughobrien** [comment]
Congratulations on your first package!

**Pull request merged**